File: apps/render-ios-playground/render-ios-playground/Application/Errors/ApplicationError.swift
/// Application-specific errors
enum ApplicationError: Error {
    case networkError(String)
    case schemaFetchFailed(String)
    case invalidResponse(String)
    case parsingError(String)
    case renderingError(String)
    
    var errorDescription: String? {
        switch self {
        case .networkError(let message):
            return "Network error: \(message)"
        case .schemaFetchFailed(let message):
            return "Schema fetch failed: \(message)"
        case .invalidResponse(let message):
            return "Invalid response: \(message)"
        case .parsingError(let message):
            return "Parsing error: \(message)"
        case .renderingError(let message):
            return "Rendering error: \(message)"
        }
    }
}

File: apps/render-android-playground/app/src/main/java/com/max_mrtnv/render_android_playground/application/errors/ApplicationError.kt
package com.max_mrtnv.render_android_playground.application.errors

/**
 * Application-specific errors
 */
sealed class ApplicationError(message: String) : Exception(message) {
    class NetworkError(message: String) : ApplicationError("Network error: $message")
    class SchemaFetchFailed(message: String) : ApplicationError("Schema fetch failed: $message")
    class InvalidResponse(message: String) : ApplicationError("Invalid response: $message")
    class ParsingError(message: String) : ApplicationError("Parsing error: $message")
    class RenderingError(message: String) : ApplicationError("Rendering error: $message")
}
